core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1192,16 +1232,15 @@ public void innerToXContent(XContentBuilder builder, Params params) throws IOExc
         }
 
         if (indexBoost != null) {
-            builder.startObject(INDICES_BOOST_FIELD.getPreferredName());
-            assert !indexBoost.containsKey(null);
-            final Object[] keys = indexBoost.keys;
-            final float[] values = indexBoost.values;
-            for (int i = 0; i < keys.length; i++) {
-                if (keys[i] != null) {
-                    builder.field((String) keys[i], values[i]);
-                }
+            builder.startArray(INDICES_BOOST_FIELD.getPreferredName());
+            assert indexBoost.stream().noneMatch(ib -> ib.v1() == null);
```

Comment:
I added a comment above, shall we rather check for null values when they get set, so we are sure this will never happen and we don't need this assertion anymore?